pint/pintk/pulsar.py

```diff
@@ -196,34 +205,151 @@ def write_fit_summary(self):
+                        getattr(self.prefit_model, param.name).frozen = getattr(self.prefit_model, 'JUMP'+str(n)).frozen
+                        self.prefit_model.remove_param('JUMP'+str(n))
+                        if self.fitted:
+                            self.postfit_model.add_param_from_top(param, 'PhaseJump')
+                            getattr(self.postfit_model, param.name).frozen = getattr(self.postfit_model, 'JUMP'+str(n)).frozen
+                            self.postfit_model.remove_param('JUMP'+str(n))
+                    if "JUMP1" not in self.prefit_model.params:
+                        #remove PhaseJump component if no jump params
+                        comp_list = getattr(self.prefit_model, 'PhaseComponent_list')
+                        for item in comp_list:
+                            if isinstance(item, pint.models.jump.PhaseJump):
+                                comp_list.remove(item)
+                                break
+                        self.prefit_model.setup_components(comp_list)
+                        if self.fitted:
+                            self.postfit_model.setup_components(comp_list)
+                    else:
+                        self.prefit_model.components["PhaseJump"].setup()
+                        if self.fitted:
+                            self.postfit_model.components["PhaseJump"].setup()
+                    print("removed param", 'JUMP'+str(num))
+                    return jump_select
+                elif True in [a and b for a,b in zip(jump_select, selected)]:
+                    #if current jump overlaps selected, raise and error and end
+                    print('Cannot jump toas that have already been jumped. Check for overlap.')
+                    return None
+        #if here, then doesn't overlap or match anything
+        for dict1, dict2 in zip(self.fulltoas.table['flags'][selected], self.toas.table['flags']):
+            dict1['jump'] = int(np.nanmax(jump_nums))+1
+            dict2['jump'] = int(np.nanmax(jump_nums))+1
+        param = pint.models.parameter.maskParameter(name = 'JUMP', index=int(np.nanmax(jump_nums))+1,\
+            key='jump', key_value = int(np.nanmax(jump_nums))+1, value = 0.0, units = 'second', aliases=['JUMP'])
+        self.prefit_model.add_param_from_top(param, "PhaseJump")
+        getattr(self.prefit_model, param.name).frozen = False
+        self.prefit_model.components["PhaseJump"].setup()
+        if self.fitted and not self.prefit_model.components["PhaseJump"] == self.postfit_model.components["PhaseJump"]:
+            self.postfit_model.add_param_from_top(param, "PhaseJump")
+            getattr(self.postfit_model, param.name).frozen = False
+            self.postfit_model.components["PhaseJump"].setup()
+        return param.name
+    
+    def fit(self, selected, iters=1):
         '''
         Run a fit using the specified fitter
         '''
+        if not any(selected):
```

Comment:
Yeah, when running a fit with no points selected, it runs the fit on all points. I put this in so that the comparison of the selected points and the existing jumps evaluates correctly.